it/test/controllers/NiphlsQuestionControllerISpec.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import helpers.ItTestBase
+import play.api.http.Status.{BAD_REQUEST, OK, SEE_OTHER}
+import play.api.libs.ws.{WSClient, WSRequest}
+import play.api.test.Helpers.{await, defaultAwaitTimeout}
+
+class NiphlsQuestionControllerISpec extends ItTestBase {
+
+  lazy val client: WSClient = app.injector.instanceOf[WSClient]
+
+  private val url = s"http://localhost:$port$appRouteContext/niphls-question"
```

Comment:
can use the controller route here to get the url instead of hard coding it
e.g.` routes.NiphlsController.onPageLoad.url`

Reply:
This is fixed

Reply:
should we not hard code it because otherwise the test is relying on something in the controller?

Reply:
@besscerule For the request to be correct it needs the correct url - if we hard code it and at some point in the future we decide to update the path for this endpoint, we would have to update the test aswell whereas this prevents that from happening as it will always get the right path for the onPageLoad function. I do see your point though as it relies us to be calling the correct function in the controller. I am happy with doing it either way as both have pros and cons. :)
val request: WSRequest = client.url(url).withFollowRedirects(false)

Reply:
happy to leave as is